src/daemon/utils_ssl.c

```diff
@@ -0,0 +1,104 @@
+/**
+ * collectd - src/utils_ssl.c
+ * Copyright (C) 2007-2010  Florian octo Forster
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; only version 2 of the License is applicable.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ * Author:
+ *   Florian octo Forster <octo at collectd.org>
+ *   Toni Moreno <toni at gmail.com>
+ */
+
+ /* Code taken from The example code from Jeremy Brown:
+ * http://curl.haxx.se/libcurl/c/opensslthreadlock.html
+ **/
+
+#if HAVE_CONFIG_H
+# include "config.h"
+#endif
+
+#ifndef HAVE_OPENSSL
+int cd_ssl_multithread_setup(void)
+{
+	return 0;
```

Comment:
This should at least log a warning and/or return non-zero (in which case the calling plugin can decide what do do).